src/main/scala/scalismo/mesh/MeshOperations.scala

```diff
@@ -54,7 +54,7 @@ class TriangleMesh3DOperations(private val mesh: TriangleMesh3D) {
   def shortestDistanceToSurfaceSquared(point: Point[_3D]): Double =
     closestPointOnSurface.getSquaredShortestDistance(point: Point[_3D])
   def closestPoint(point: Point[_3D]): ClosestPoint = closestPointOnSurface.getClosestPoint(point)
-  def closestPointOnSurface(point: Point[_3D]): ClosestPointOnSurface =
+  def closestPointOnSurface(point: Point[_3D]): ClosestPointSpecialized =
```

Comment:
How about ```ClosestPointWithType``` instead of specialized? Later on you use the name ClosestPointType for this distinction.

Reply:
Thank you for this suggestion. I like this name.